### PR TITLE
Details

### PR DESCRIPTION
**Pull request by pascalweber85**
i add a detailspage to the app and make it works all together

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,17 +1,39 @@
...
+  const [activePage, setActivePage] = useState('LocationPage')
+  const [detailPage, setDetailPage] = useState(null)
+
   return (
     <AppGrid>
-      <LocationPage />
+      {activePage === 'LocationPage' && <LocationPage onDetail={showDetail} />}
+
+      {activePage === 'DetailsPage' && (
+        <DetailsPage
```
Comment (Morten-Moeller) [inline src/App.js]: At this point you just want to hand over the detailPage as detailPage and then de-structure it on the DetailsPage.
If you have questions, feel free to contact me.

File: src/App.js [hunk 2]
```diff
@@ -1,17 +1,39 @@
...
+          öffnungszeiten={detailPage.öffnungszeiten}
+          beschreibung={detailPage.beschreibung}
+        />
+      )}
     </AppGrid>
   )
+
+  function showDetail(title, image, öffnungszeiten, beschreibung) {
+    setActivePage('DetailsPage')
+    setDetailPage({ title, image, öffnungszeiten, beschreibung })
```
Comment (Morten-Moeller) [inline src/App.js]: is there only one detailPage? Because you override your detailPage here, with just one.

File: src/components/Button.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import PropTypes from 'prop-types'
+import styled from 'styled-components/macro'
+
+Button.propTypes = {
+  onClick: PropTypes.func.isRequired,
+  isActive: PropTypes.bool,
+  children: PropTypes.node,
+}
+
+export default function Button(props) {
```
Comment (Morten-Moeller) [inline src/components/Button.js:10]: You can export just the styled.button, that's cleaner and you don't have to check for the props

File: src/pages/LocationPage.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+LocationPage.propTypes = {
+  onDetail: PropTypes.func.isRequired,
+}
+
+export default function LocationPage({ onDetail }) {
+  return (
+    <>
+      <Header>Best Locations</Header>
+      <Wrapper>
+        <div>
```
Comment (Morten-Moeller) [inline src/pages/LocationPage.js]: Do you really need this div?

**Review by Morten-Moeller — COMMENTED** [review]
Maybe we should meet and talk about state. 👾
Keep on rolling!

File: src/App.js [hunk 5]
```diff
@@ -2,30 +2,30 @@ import { useState } from 'react'
 import styled from 'styled-components/macro'
 import LocationPage from './pages/LocationPage'
 import DetailsPage from './pages/DetailsPage'
+import data from './data.json'
 
 export default function App() {
   const [activePage, setActivePage] = useState('LocationPage')
```
Comment (hugh-burgess) [inline src/App.js:8]: You could put all these states into a use custom hook?

File: src/data.json [hunk 6]
```diff
@@ -171,6 +258,11 @@
     "text": "€€€€, Club & Disco",
     "main_image_path": "https://s3-media0.fl.yelpcdn.com/bphoto/bow4ZZU5X0Y6Fj4fsMuAYA/348s.jpg",
     "beschreibung": "Darkflower Alternative Club Leipzig- Das Darkflower ist Treff der schwarzen Szene. Der bekannteste schwarze Szene-Club bietet alles von Alternative bis EBM.",
-    "öffnungszeiten": "Montag 10 - 18 Uhr Dienstag 10 - 18 Uhr Mittwoch 10 - 18 Uhr Donnerstag 10 - 18 Uhr"
+    "öffnungszeiten": [
+      "Montag 10 - 18 Uhr",
+      "Dienstag 10 - 18 Uhr",
+      "Mittwoch 10 - 18 Uhr",
+      "Donnerstag 10 - 18 Uhr"
+    ]
```
Comment (hugh-burgess) [inline src/data.json:342]: I think you’d save some lines of you could group the object into a smaller chunk. Maybe as a single object that you call? Not sure though.

**Review by hugh-burgess — COMMENTED** [review]
LGTM man, couple things. I think you could look up how to squash a repetitive chunk of code into something smaller. But great work.

**Pull request merged**